tests/ui/redundant_test_prefix.rs

```diff
@@ -0,0 +1,159 @@
+#![allow(dead_code)]
+#![warn(clippy::redundant_test_prefix)]
+
+fn main() {
+    // Normal function, no redundant prefix.
+}
+
+fn f1() {
+    // Normal function, no redundant prefix.
+}
+
+fn test_f2() {
+    // Has prefix, but no `#[test]` attribute, ignore.
+}
+
+#[test]
+fn test_f3() {
+    // Has prefix, has `#[test]` attribute.
+    // Not within a `#[cfg(test)]`, ignore (by default, configurable -- see TOML tests).
+}
+
+#[cfg(test)]
+#[test]
+fn test_f4() {
+    //~^ ERROR: redundant `test_` prefix in test function name
```

Comment:
If the same error and help messages are given, it is not really important to check them in the `.rs` file, as they will show up in the `.stderr` file and a change would require a new blessing. Maybe you could just use `//~ redundant_test_prefix` in the test files.